### PR TITLE
DOCS-1781: Handle planning for the zero pose

### PR DESCRIPTION
**Pull request by npentrel**

### CODE REVIEW

File: docs/mobility/motion/_index.md [hunk 1]
```diff
@@ -358,26 +358,29 @@ logger.Info("Orientation of myArm from the motion service:", myArmMotionPose.Pos
...
 
 Move a [base](/components/base/) component to a destination [`Pose`](https://python.viam.dev/autoapi/viam/proto/common/index.html#viam.proto.common.Pose) on a {{< glossary_tooltip term_id="slam" text="SLAM" >}} map.
-Use the machine's position reported by the {{< glossary_tooltip term_id="slam" text="SLAM" >}} service to check the location of the machine.
-
-{{< alert title="Usage" color="tip" >}}
 
 `MoveOnMap()` is non blocking, meaning the motion service will move the component to the destination [`Pose`](https://python.viam.dev/autoapi/viam/proto/common/index.html#viam.proto.common.Pose) after `MoveOnMap()` returns.
 
-Each successful `MoveOnMap()` call retuns a unique `ExecutionID` which you can use to identify all plans generated during the `MoveOnMap()` call.
+Each successful `MoveOnMap()` call returns a unique `ExecutionID` which you can use to identify all plans generated during the `MoveOnMap()` call.
```
Comment (nfranczak) [inline docs/mobility/motion/_index.md:364]: This also applies to `MoveOnGlobe()`. May you please update that as well?

File: docs/mobility/motion/_index.md [hunk 2]
```diff
@@ -474,25 +477,27 @@ executionID, err := motionService.MoveOnMap(context.Background(), motion.MoveOnM
 Move a [base](/components/base/) component to a destination GPS point, represented in geographic notation _(latitude, longitude)_.
 Use a [movement sensor](/components/movement-sensor/) to check the location of the machine.
 
-{{< alert title="Usage" color="tip" >}}
-
 `MoveOnGlobe()` is non blocking, meaning the motion service will move the component to the destination GPS point after `MoveOnGlobe()` returns.
 
 Each successful `MoveOnGlobe()` call retuns a unique `ExecutionID` which you can use to identify all plans generated during the `MoveOnGlobe()`.
```
Comment (npentrel) [inline docs/mobility/motion/_index.md:482]:
```suggestion
Each successful `MoveOnGlobe()` call returns a unique `ExecutionID` which you can use to identify all plans generated during the `MoveOnGlobe()`.
```

File: docs/mobility/motion/_index.md (hunk 1 above)
Comment (npentrel) [reply docs/mobility/motion/_index.md:364]: done, thanks

**Review by nfranczak — APPROVED** [review]

File: docs/mobility/motion/_index.md [hunk 3]
```diff
@@ -358,26 +358,29 @@ logger.Info("Orientation of myArm from the motion service:", myArmMotionPose.Pos
...
-
-{{< alert title="Usage" color="tip" >}}
 
 `MoveOnMap()` is non blocking, meaning the motion service will move the component to the destination [`Pose`](https://python.viam.dev/autoapi/viam/proto/common/index.html#viam.proto.common.Pose) after `MoveOnMap()` returns.
 
-Each successful `MoveOnMap()` call retuns a unique `ExecutionID` which you can use to identify all plans generated during the `MoveOnMap()` call.
+Each successful `MoveOnMap()` call returns a unique `ExecutionID` which you can use to identify all plans generated during the `MoveOnMap()` call.
+
+{{< alert title="Info" color="info" >}}
+If you specify a goal pose or orientation and the robot's current position is already at the goal post or orientation, `MoveonMap` returns an error.
```
Comment (nfranczak) [inline docs/mobility/motion/_index.md:367]: small nit: I would say:
```
If you specify a goal pose and the robot's current position is already within the set `PlanDeviationM`, then `MoveonMap` returns an error.
```

i.e. if we are within PlanDeviationMM of the goal pose then we do not need to start.

**Review by nfranczak — CHANGES_REQUESTED** [review]
nit change `is already at the goal post or orientation` to `within planDevM` for both moveonmap and moveonglobe

File: docs/mobility/motion/_index.md (hunk 3 above)
Comment (npentrel) [inline docs/mobility/motion/_index.md:367]:
```suggestion
If you specify a goal pose and the robot's current position is already within the set `PlanDeviationM`, then `MoveonMap` returns an error.
```

File: docs/mobility/motion/_index.md [hunk 4]
```diff
@@ -474,25 +477,27 @@ executionID, err := motionService.MoveOnMap(context.Background(), motion.MoveOnM
...
 
-{{< alert title="Usage" color="tip" >}}
-
 `MoveOnGlobe()` is non blocking, meaning the motion service will move the component to the destination GPS point after `MoveOnGlobe()` returns.
 
-Each successful `MoveOnGlobe()` call retuns a unique `ExecutionID` which you can use to identify all plans generated during the `MoveOnGlobe()`.
+Each successful `MoveOnGlobe()` call returns a unique `ExecutionID` which you can use to identify all plans generated during the `MoveOnGlobe()`.
+
+{{< alert title="Info" color="info" >}}
+If you specify a goal pose or orientation and the robot's current position is already at the goal post or orientation, `MoveonMap` returns an error.
```
Comment (npentrel) [inline docs/mobility/motion/_index.md:485]:
```suggestion
If you specify a goal pose or orientation and the robot's current position is already within the set `PlanDeviationM` or orientation, `MoveOnGlobe` returns an error.
```

File: docs/mobility/motion/_index.md [hunk 5]
```diff
@@ -474,25 +477,27 @@ executionID, err := motionService.MoveOnMap(context.Background(), motion.MoveOnM
...
 
-{{< alert title="Usage" color="tip" >}}
-
 `MoveOnGlobe()` is non blocking, meaning the motion service will move the component to the destination GPS point after `MoveOnGlobe()` returns.
 
-Each successful `MoveOnGlobe()` call retuns a unique `ExecutionID` which you can use to identify all plans generated during the `MoveOnGlobe()`.
+Each successful `MoveOnGlobe()` call returns a unique `ExecutionID` which you can use to identify all plans generated during the `MoveOnGlobe()`.
+
+{{< alert title="Info" color="info" >}}
+If you specify a goal pose or orientation and the robot's current position is already within the set `PlanDeviationM` or orientation, `MoveOnGlobe` returns an error.
```
Comment (nfranczak) [inline docs/mobility/motion/_index.md:485]: can you please change this to be
```
If you specify a goal pose and the robot's current position is already within the set `PlanDeviationM`,  `MoveOnGlobe` returns an error.
```

**Comment by npentrel** [comment]
@nfranczak is this now good to go?

File: docs/mobility/motion/_index.md (hunk 5 above)
Comment (nfranczak) [reply docs/mobility/motion/_index.md:485]: @npentrel this should not include `"or orientation"`

**Review by nfranczak — APPROVED** [review]
One comment about removing `"or orientation"` from `MoveOnGlobe`. Otherwise, LGTM

File: docs/mobility/motion/_index.md (hunk 5 above)
Comment (npentrel) [inline docs/mobility/motion/_index.md:485]:
```suggestion
If you specify a goal pose and the robot's current position is already within the set `PlanDeviationM`, `MoveOnGlobe` returns an error.
```

**Pull request merged**